git/revlist.go

```diff
@@ -117,3 +120,207 @@ func revListCallback(c *Client, opt RevListOptions, commits []CommitID, excludeL
 	}
 	return nil
 }
+
+// Internal pre-computed record of which symbolic references
+//  are pointing to a particular ref name (without the refs/ prefix)
+var symRefsForRefName map[string]string
+
+// Internal pre-computed record of which ref names are pointing to
+//  to a particular commit. Values are in the form of "HEAD -> master, branch2,
+//  tag: tag1"
+var refNamesForCommit map[string]string
+
+// We have been asked to pre-compute the ref names for important
+//  commits as a way to improve performance.
+func buildRefNames(c *Client) error {
```

Comment:
Can most of this function be replaced with a call to ShowRef? If so, it might be a good idea to use the git plumbing that's already there.

Reply:
Sure, I'll have a look. I wasn't sure if we had something that calculated the ref names of a commit in dgit already.